lib/hypervisor/hv_kvm/__init__.py

```diff
@@ -1964,17 +1969,27 @@ def _ExecuteKVMRuntime(self, instance, kvm_runtime, kvmhelp, incoming=None):
           vhostfd = ""
 
         if kvm_supports_netdev:
+          dev_ops = []
+
           # Non paravirtual NICs hvinfo is empty
           if "id" in nic.hvinfo:
             nic_val = _GenerateDeviceHVInfoStr(nic.hvinfo)
             netdev = nic.hvinfo["id"]
           else:
             nic_val = "%s" % nic_model
             netdev = "netdev%d" % nic_seq
+
           nic_val += (",netdev=%s,mac=%s%s" % (netdev, nic.mac, nic_extra))
           tap_val = ("type=tap,id=%s,%s%s%s" %
                      (netdev, tapfd, vhostfd, tap_extra))
-          kvm_cmd.extend(["-netdev", tap_val, "-device", nic_val])
+          dev_ops.append(nic_val)
+
+          # add bootindex property to the first nic if network boot is enabled
+          if boot_network and incoming is None:
```

Comment:
what's the idea behind "not when `incoming`"? will then a network booted instance not be able to reboot after live migration? maybe "not when `kernel_path` was meant?